homeassistant/components/mopeka/config_flow.py

```diff
@@ -89,6 +123,49 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user",
             data_schema=vol.Schema(
-                {vol.Required(CONF_ADDRESS): vol.In(self._discovered_devices)}
+                {
+                    vol.Required(CONF_ADDRESS): vol.In(self._discovered_devices),
+                    vol.Required(
+                        CONF_MEDIUM_TYPE,
+                        default=MediumType.PROPANE.value,
+                    ): vol.In({medium.value: medium.name for medium in MediumType}),
+                }
+            ),
+        )
+
+
+class MopekaOptionsFlow(config_entries.OptionsFlow):
+    """Handle options for the Mopeka component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle options flow."""
+        if user_input is not None:
+            new_data = {
+                **self.config_entry.data,
+                CONF_MEDIUM_TYPE: user_input[CONF_MEDIUM_TYPE],
+            }
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, data=new_data
+            )
+            await self.hass.config_entries.async_reload(self.config_entry.entry_id)
+            return self.async_create_entry(title="", data={})
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_MEDIUM_TYPE,
+                        default=self.config_entry.data.get(
+                            CONF_MEDIUM_TYPE, MediumType.PROPANE.value
+                        ),
+                    ): vol.In({m.value: m.name for m in MediumType}),
```

Comment:
The names should be formatted for humans

<img width="330" alt="Screenshot 2024-07-29 at 5 49 18 PM" src="https://github.com/user-attachments/assets/34ce2d85-1e16-40d8-ace3-dcbeb402158c">

Something simple like `m.name.replace("_"," ").title()` (or make it a function since it needs to be used in multiple places)